td-layout/src/lib.rs

```diff
@@ -42,6 +42,53 @@ pub const MIN_MEMORY_SIZE: u64 =
 
 pub const TD_PAYLOAD_PARTIAL_ACCEPT_MEMORY_SIZE: u32 = 0x10000000;
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[repr(u32)]
+pub enum E820Type {
```

Comment:
why E820 type is related layout ???

Reply:
then we can generate E820 table at runtime based on layout config automatically